consensus/core/src/core.rs

```diff
@@ -304,6 +307,28 @@ impl Core {
             }
         }
 
+        let leader_authority = &self
+            .context
+            .committee
+            .authority(self.first_round_leader(clock_round))
```

Comment:
I believe we need the previous round here, hence something like `clock_round.saturating_sub(1)`